accounts/backends.py

```diff
@@ -49,11 +50,12 @@ def authenticate(self, request, remote_user, shibboleth_attributes):
             if key != "affiliation" and getattr(user, key) is not value:
                 setattr(user, key, value)
 
-        # Update affiliations with every log in
-        user.affiliation.clear()
+        # Update groups with every log in
+        user.groups.clear()
         for affiliation_name in shibboleth_attributes["affiliation"]:
-            affiliation, _ = PennAffiliation.objects.get_or_create(name=affiliation_name)
-            user.affiliation.add(affiliation)
+            if affiliation_name:  # Some users don't have any affiliation somehow ¯\_(ツ)_/¯
```

Comment:
would explicitly say `if affiliation_name is not None` –– don't like relying on falsiness in Python.

Reply:
Makes sense